client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,198 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import page from 'page';
+import DomainPicker from '@automattic/domain-picker';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { ThemeProvider } from 'emotion-theming';
+import styled from '@emotion/styled';
+import { translate } from 'i18n-calypso';
+import classnames from 'classnames';
+import { isDesktop } from '@automattic/viewport';
+import type { DomainSuggestions } from '@automattic/data-stores';
+
+/**
+ * Internal dependencies
+ */
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import {
+	MARKETPLACE_FLOW_ID,
+	ANALYTICS_UI_LOCATION_MARKETPLACE_DOMAIN_SELECTION,
+} from 'calypso/my-sites/plugins/marketplace/constants';
+import { getWpComDomainBySiteId } from 'calypso/state/sites/domains/selectors';
+import { fetchSiteDomains } from 'calypso/state/sites/domains/actions';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { getProductsList, isProductsListFetching } from 'calypso/state/products-list/selectors';
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+import Masterbar from 'calypso/layout/masterbar/masterbar';
+import Item from 'calypso/layout/masterbar/item';
+import getPreviousPath from 'calypso/state/selectors/get-previous-path';
+import { HorizontalRule } from '@wordpress/components';
+import ExternalLink from 'calypso/components/external-link';
+import MarketplaceShoppingCart from 'calypso/my-sites/plugins/marketplace/components/marketplace-shopping-cart';
+import theme from 'calypso/my-sites/plugins/marketplace/theme';
+
+/**
+ * Style dependencies
+ */
+import 'calypso/my-sites/plugins/marketplace/marketplace-domain-upsell/style.scss';
+
+const MarketplaceHeaderTitle = styled.h1`
+	font-size: 2rem;
+	margin-bottom: 8px;
+`;
+const MarketplaceHeaderSubTitle = styled.h2`
+	font-size: 0.875rem;
+	margin-bottom: 15px;
+`;
+
+function MarketplaceDomainUpsellHeader() {
+	return (
+		<div>
+			<MarketplaceHeaderTitle className="marketplace-domain-upsell__title wp-brand-font">
+				{ translate( 'Choose a domain' ) }
```

Comment:
Did we intend to use `translate` directly here? I'd expect us to be using the `useTranslate` hook instead. Otherwise we're risking to remain with the old strings in the event of a locale change.

Reply:
Didnt know we had a hook, however aren't these practically static strings? When would this text change in prod?

Reply:
Those are localized and will display a translated version, considering the current user's locale. When the user changes their interface language, the locale would change and the strings could change to be the translated versions of those strings in the new language.

Reply:
Ah I see got it! :)